example-gto/code/mlem/.github/workflows/gto-act-on-tags.yml

```diff
@@ -0,0 +1,32 @@
+name: Act on artifact registrations and promotions
+on:
+  push:
+    tags:
+      - "*"
+env:
+  HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
+
+jobs:
+  act:
+    name: Figure out what was registered/promoted and act on it
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: "GTO: figure out what was registered/promoted and show the Registry state"
+      id: gto
+      uses: iterative/gto-action@v1
+      with:
+        show: true
+        history: true
+    - uses: actions/setup-python@v2
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip setuptools wheel
+        pip install -r requirements.txt
+    - name: "Deploy a MLEM model (act on assigning a new stage)"
+      if: steps.gto.outputs.event == 'assignment'
+      run: |
+        echo "[$GITHUB_REF] You got model '${{ steps.gto.outputs.name }}' of version '${{ steps.gto.outputs.version }}' promoted to stage '${{ steps.gto.outputs.stage }}'"
+        mlem deploy run deploy/${{ steps.gto.outputs.stage }}
```

Comment:
Q: how will we connect deployments to stages in Studio?

Reply:
Q: can we deploy an older model (from an older commit) to the newer service code (main HEAD)?

Reply:
> Q: can we deploy an older model (from an older commit) to the newer service code (main HEAD)?

There is a similar question raised by @mike0sv. We can discuss it there https://github.com/iterative/gto/issues/245

Reply:
> Q: how will we connect deployments to stages in Studio?

I guess there is no way to do that in this "happy path". But if you're going to create a MLEM config that connects GTOStages + ModelName with MLEMDeployments, than it should be possible I think. WDYT @mike0sv?

Reply:
We discussed this with @shcheklein, it's possible to add this config with addition of something like `mlem deploy run --config mlem-gto-mapping.yaml`. Not sure we should add this in 0.3.0 though
Another idea is to exclude `model` field from deployment declaration, so you always need to specify model in `mlem deploy run`